hibernate-core/src/main/java/org/hibernate/sql/results/graph/entity/internal/AbstractBatchEntitySelectFetchInitializer.java

```diff
@@ -75,6 +75,10 @@ protected void resolveKey(
 			RowProcessingState rowProcessingState,
 			ToOneAttributeMapping referencedModelPart,
 			FetchParentAccess parentAccess) {
+		if ( parentAccess.findFirstEntityInitializer().isEntityInitialized() ) {
+			state = State.INITIALIZED;
```

Comment:
Can you please add a code comment explaining this? I don't really understand why this is necessary. Also, why not move it down after the `if ( state != State.UNINITIALIZED ) {` block, given that `parentAccess.findFirstEntityInitializer()` might require some pointer chasing?